tests/func/test_lock.py

```diff
@@ -1,23 +1,23 @@
-import os
+import pytest
 
-from dvc.lock import Lock, LockError
+from dvc.lock import FAILED_TO_LOCK_MESSAGE, Lock, LockError
 from dvc.main import main
-from tests.basic_env import TestDvc
 
 
-class TestLock(TestDvc):
-    def test_with(self):
-        lockfile = os.path.join(self.dvc.dvc_dir, "tmp", "lock")
-        lock = Lock(lockfile)
-        with lock:
-            with self.assertRaises(LockError):
-                lock2 = Lock(lockfile)
-                with lock2:
-                    pass
+def test_with(tmp_dir, dvc, mocker):
+    # patching to speedup tests
+    mocker.patch("dvc.lock.DEFAULT_TIMEOUT", 0.01)
```

Comment:
Reduced it further to 0.01.

Reply:
So maybe let's just adjust it in the code as discussed above?

Reply:
This patch will be required after that too unless we decrease the timeout to a very low value.

Let's say we reduced the timeout to 2s. As these tests are negative tests, they will wait for 2s anyway.

Reply:
@skshetry Good point, so we will have to patch both timeout and the number of tries.